beta-operator-rewards/rewards-requirements.ts

```diff
@@ -0,0 +1,229 @@
+import { Contract } from "ethers";
+import { program } from "commander";
+import * as fs from "fs";
+import { ethers } from "ethers";
+import {
+  abi as RandomBeaconABI,
+  address as RandomBeaconAddress,
+} from "@keep-network/random-beacon/artifacts/RandomBeacon.json";
+import {
+  abi as WalletRegistryABI,
+  address as WalletRegistryAddress,
+} from "@keep-network/ecdsa/artifacts/WalletRegistry.json";
+import axios from "axios";
+
+export async function calculateBetaOperatorFactors() {
+  program
+    .version("0.0.1")
+    .requiredOption(
+      "-s, --start <timestamp>",
+      "starting time for rewards calculation"
+    )
+    .requiredOption(
+      "-e, --end <timestamp>",
+      "ending time for rewards calculation"
+    )
+    .requiredOption("-i, --interval <timestamp>", "scrape interval") // IMPORTANT, it has to be the same as in prometheous config
+    .requiredOption(
+      "-a, --api <prometheus api>",
+      "prometheus API"
+    )
+    .requiredOption(
+      "-j, --job <prometheus job>",
+      "prometheus job"
+    )
+    .requiredOption(
+      "-o, --output <file>",
+      "output JSON file"
+    )
+    .parse(process.argv);
+
+  const options = program.opts();
+
+  const queryStep = 600; // 10min in sec
+  const prometheous_job = options.job;
+  const prometheusAPI = options.api
+  const startRewardsTimestamp = parseInt(options.start);
+  const endRewardsTimestamp = parseInt(options.end);
+  const scrapeInterval = parseInt(options.interval);
+  const rewardsInterval = endRewardsTimestamp - startRewardsTimestamp;
+  const requiredUptime = 96; // percent
+  const preParamsAvgInterval = 24 // hours
+  const preParamsResolution = 24 // hours
+  const minPreParams = 500 // min requirement for pre params daily avg
+  const factors = {
+    isBeaconAuthorized: "isBeaconAuthorized",
+    isTbtcAuthorized: "isTbtcAuthorized",
+    upTime: "upTime",
+    version: "verion",
+    preParams: "preParams",
+  };
+  const upTimeRewardsCoefficient = "upTimeRewardsCoefficient"
+  const peersDataFile = options.output
```

Comment:
I would group together all of the options parsing into tight block

Reply:
Sorry, I'm not sure I'm following, can you pls give an example?

Reply:
sure!

```typescript
  // Parse the program options
  const options = program.opts();
  const prometheus_job = options.job;
  const prometheusAPI = options.api;
  const clientVersions = options.versions;
  const startRewardsTimestamp = parseInt(options.start);
  const endRewardsTimestamp = parseInt(options.end);
  const scrapeInterval = parseInt(options.interval);
  const peersDataFile = options.output;
  // End program option parsing
```

Reply:
Addressed in 734c3b4e9cd57d5a03e29b90e1037a3a4ec4ae33

Reply:
Addressed in 734c3b4e9cd57d5a03e29b90e1037a3a4ec4ae33